### ISSUE TITLE
[Bug]: Open Dialog on Ubuntu 22 has a "Open files read-only" checkbox

### ISSUE THREAD

**Issue by gtritchie** [electron/electron#42349]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

30.0.9

### What operating system are you using?

Ubuntu

### Operating System Version

Ubuntu 22.04.4 LTS

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

30.0.3

### Expected Behavior

Invoke `dialog.ShowOpenDialog()`; the dialog has a "Open files read-only" checkbox in the lower left.

<img width="861" alt="screenshot of Open File dialog on Ubuntu 22 with a open file read-only checkbox" src="https://github.com/electron/electron/assets/10569626/1c342215-7248-4014-b515-c4f07d3f04e9">

### Actual Behavior

No such checkbox, or at least, a way to suppress it since this isn't a useful option in our scenario (and doesn't seem to be any way to tell if the user checked it even if it was).

**Comment by codebytere** [comment]
@gtritchie is this new or has it been present in previous versions? My understanding is it's a Flatpak feature we don't really have control over: https://github.com/flatpak/xdg-desktop-portal/issues/808

**Comment by gtritchie** [comment]
Apologies for not providing that detail initially; this first happened in 30.0.4; in 30.0.3 and earlier the dialog looked like this:

<img width="928" alt="older-dialog" src="https://github.com/electron/electron/assets/10569626/d3170860-9242-48ce-80d7-47b917aab3bd">

Presumably related to the fix for https://github.com/electron/electron/issues/32857.

**Comment by codebytere** [comment]
@gtritchie yep, that's what i was checking :) I'll take a look!

**Comment by codebytere** [comment]
@gtritchie this unfortunately looks like a consequence of https://github.com/electron/electron/pull/42045, but not one that I think we can fix on our end. Previously, we only had a GTK dialog implementation, and now there's a separate `SelectFileDialogLinuxPortal` [implementation](https://source.chromium.org/chromium/chromium/src/+/main:ui/shell_dialogs/select_file_dialog_linux_portal.h?q=portal%20dialog%20linux&sq=package:chromium&type=cs), which is why you're seeing this. As far as I can tell there's not a way to suppress that but if that changes we can revisit this!

**Comment by Mikenux** [comment]
Hello @codebytere and @gtritchie!

I saw that you mentioned this issue in https://github.com/flatpak/xdg-desktop-portal/issues/808.

The (unofficial) plan is to remove this checkbox and have a write access portal instead. I recently opened a discussion where cases can be reported: https://github.com/flatpak/xdg-desktop-portal/discussions/1397.

So, one may wonder if this checkbox works: https://github.com/flatpak/xdg-desktop-portal-gtk/issues/433. Note that not being a developer, I don't know if I missed something.
